### PR TITLE
ROC-5750:See this courts facilities

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-5750

### Change description
Show court facilities in hearing location page

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/app/court-finder/courtFinderClient.ts [hunk 1]
```diff
@@ -24,6 +26,16 @@ export class CourtFinderClient {
     return this.performRequest(uri)
   }
 
+  public getCourtDetails (slug: string): Promise<CourtDetailsResponse> {
+    if (!slug) {
+      return Promise.reject('Missing slug')
+    }
+
+    let uri: string = `${this.apiUrl}/court-finder/court-details/${slug}`
```
Comment (robertparkinson) [inline src/main/app/court-finder/courtFinderClient.ts]: Can you change this to a const please?

File: src/main/features/directions-questionnaire/routes/hearing-location.ts [hunk 2]
```diff
@@ -54,19 +63,27 @@ export default express.Router()
   .get(Paths.hearingLocationPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
     try {
       const draft: Draft<DirectionsQuestionnaireDraft> = res.locals.draft
+      //this is not right place - should be moved to tasklist builder
+      draft.document.hearingLocation = undefined
```
Comment (robertparkinson) [inline src/main/features/directions-questionnaire/routes/hearing-location.ts]: should this be moved or removed?

File: src/main/features/directions-questionnaire/views/hearing-location.njk [hunk 3]
```diff
@@ -29,8 +30,19 @@
             {% elseif party === MadeBy.CLAIMANT %}
               <p>{{ t('You can request any location - {{ courtName }} is the nearest to your home address you gave us.', { courtName: courtName }) }}
             {% endif %}
-            <h1
-              class="heading-medium">{{ t('Do you want to choose {{ courtName }}?', { courtName: form.valueFor('courtName') }) }}</h1>
+
+            <details class="form-group analytics-click-event-trigger" data-event-label="Response: Find out what each response means">
```
Comment (andrewwa-kainos) [inline src/main/features/directions-questionnaire/views/hearing-location.njk]: Label has been copied - should be updated

File: src/main/features/directions-questionnaire/views/hearing-location.njk [hunk 4]
```diff
@@ -29,8 +30,19 @@
...
+
+              <div class="panel">
+                {% for facility in courtFacilities %}
+                  <h3 class="heading-small form-group-related">{{ facility.name }}</h3>
+                  {{ facility.description | safe }}
+                {% endfor %}
+              </div>
+            </details>
+
+            <h1 class="heading-medium">{{ t('Do you want to choose {{ courtName }}', { courtName: form.valueFor('courtName') }) }}</h1>
```
Comment (andrewwa-kainos) [inline src/main/features/directions-questionnaire/views/hearing-location.njk]: Prototype has a question mark

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 2 above)
Comment (KiranVarma02) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: Yes, removed or moved or may be it's the place. When I was working on this, I didn't have complete context so I was just trying to get this page to work. Please speak to @sudheer or someone who knows DQ journey if you have questions

File: src/main/features/directions-questionnaire/views/hearing-location.njk (hunk 4 above)
Comment (karthikkannankainos) [reply src/main/features/directions-questionnaire/views/hearing-location.njk]: done

File: src/main/features/directions-questionnaire/views/hearing-location.njk (hunk 3 above)
Comment (karthikkannankainos) [reply src/main/features/directions-questionnaire/views/hearing-location.njk]: Done

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 2 above)
Comment (karthikkannankainos) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: @sudheer-ajax let we discuss about this.

File: src/main/app/court-finder/courtFinderClient.ts (hunk 1 above)
Comment (karthikkannankainos) [reply src/main/app/court-finder/courtFinderClient.ts]: done

File: src/main/features/directions-questionnaire/routes/hearing-location.ts (hunk 2 above)
Comment (karthikkannankainos) [reply src/main/features/directions-questionnaire/routes/hearing-location.ts]: After discussing with @sudheer-ajax , have removed this statement to set undefined explicitly and have removed the following if else condition as well. Note - @KiranVarma02 , @robertparkinson

**Pull request merged**